File: client-domain/src/main/java/com/jraska/github/client/time/RealTimeProvider.java
package com.jraska.github.client.time;

import java.util.concurrent.TimeUnit;

public final class RealTimeProvider implements TimeProvider {
  public static final RealTimeProvider INSTANCE = new RealTimeProvider();

  private RealTimeProvider() {
  }

  @Override public long elapsed() {
    return TimeUnit.NANOSECONDS.toMillis(System.nanoTime());
  }
}

File: client-domain/src/main/java/com/jraska/github/client/time/RealTimeProvider.kt
package com.jraska.github.client.time

import java.util.concurrent.TimeUnit

class RealTimeProvider private constructor() : TimeProvider {

    override fun elapsed(): Long {
        return TimeUnit.NANOSECONDS.toMillis(System.nanoTime())
    }

    companion object {
        val INSTANCE = RealTimeProvider()
    }
}
